plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -0,0 +1,317 @@
+
+    // App Engine Project ID
+    Label projectIdLabel = new Label(container, SWT.NONE);
+    projectIdLabel.setText("App Engine Project ID: (optional)"); //$NON-NLS-1$
+    projectIdField = new Text(container, SWT.BORDER);
+    GridData projectIdPosition = new GridData(GridData.FILL_HORIZONTAL);
+    projectIdPosition.horizontalSpan = 2;
+    projectIdField.setLayoutData(projectIdPosition);
+    projectIdField.addModifyListener(pageValidator);
+  }
+
+  protected void openLocationDialog() {
+    DirectoryDialog dialog = new DirectoryDialog(getShell());
+    dialog.setText("Please select the location to contain generated project");
+    String loc = dialog.open();
+    if (loc != null) {
+      locationField.setText(loc);
+      checkPageComplete();
+    }
+  }
+
+  protected void checkPageComplete() {
+    setPageComplete(validatePage());
+  }
+
+  /**
+   * Validate and report on the contents of this page
+   * 
+   * @return true if valid, false if there is a problem
+   */
+  public boolean validatePage() {
+    setMessage(null);
+    setErrorMessage(null);
+    
+    // order here should match order of the UI fields
+
+    String location = locationField.getText().trim();
+    if (!useDefaults() && location.isEmpty()) {
+      setMessage("Please provide a location", INFORMATION);
+      return false;
+    }
+
+    if (!validateMavenSettings()) {
+      return false;
+    }
+    if (!validateGeneratedProjectLocation()) {
+      return false;
+    }
+
+    String packageName = getPackageName();
+    if (!JavaPackageValidator.validate(packageName)) {
+      setErrorMessage(MessageFormat.format("Illegal Java package name: {0}.", packageName));
+      return false;
+    }
+
+    String projectId = getAppEngineProjectId();
+    if (!AppEngineProjectIdValidator.validate(projectId)) {
+      setErrorMessage(MessageFormat.format("Illegal App Engine Project ID: {0}.", projectId));
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Check that we won't overwrite an existing location. Expects a valid Maven Artifact Id.
+   */
+  private boolean validateGeneratedProjectLocation() {
+    String artifactId = getArtifactId();
+    // assert !artifactId.isEmpty()
+    IPath path = getLocationPath().append(artifactId);
+    if (path.toFile().exists()) {
+      setErrorMessage(MessageFormat.format("Location already exists: {0}.", path));
+      return false;
+    }
+    return true;
+  }
+
+  private boolean validateMavenSettings() {
+    String groupId = getGroupId();
+    if (groupId.isEmpty()) {
+      setMessage("Please provide Maven Group Id.", INFORMATION);
+      return false;
+    } else if (!MavenCoordinatesValidator.validateGroupId(groupId)) {
+      setErrorMessage(MessageFormat.format("Illegal Maven Group Id: {0}.", groupId));
+      return false;
+    }
+    String artifactId = getArtifactId();
+    if (artifactId.isEmpty()) {
+      setMessage("Please provide Maven Artifact Id.", INFORMATION);
```

Comment:
I'm going to leave these "Maven Artifact Id" as is to mirror the M2E uses.  It also helps distinguish the App Engine Project ID, since Maven also talks about projects.